### ISSUE TITLE
Factory crashing on update to 3.10.6 because submit_attrs.cfg is not created when there are no submit attributes

### ISSUE THREAD

**Issue by mmascher** [glideinWMS/glideinwms#388]
The actual error is that `submit_attrs.cfg` files are not created in the work-dir and entries subdirectories when upgrading an existing Factory and in absence of content to put in the files (no submit_attr entries for that section -- global or entry config). The missing files cause the Factory to crash.
See the comment below for a test and a workaround.

ISSUE EDITED. THIS ISSUE WAS:  "Entries might be configured with no submit attributes"

Still related to this https://github.com/glideinWMS/glideinwms/pull/382

After applying the fix from #387 I hit another road blocker. It turns out that not all the entries have submit attributes, and when an entry does not have any submit attribute a submit_attrs.cfg file is not created. In order to conrtinue with the tests in ITB what I did was to run this command:

```
[mmascher@vocms0205 work-dir]$ pwd
/var/lib/gwms-factory/work-dir
[mmascher@vocms0205 work-dir]$ for dir in `find . -maxdepth 1 -type d`; do sudo -u gfactory touch $dir/submit_attrs.cfg; done
```

The exception I was hitting was:

```
[2024-01-16 16:11:48,506] WARNING: glideFactory:413: EntryGroup 0 STDERR: b'Traceback (most recent call last):\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryEntryGroup.py", line 743, in <module>\n    main(int(sys.argv[1]), int(sys.argv[2]), int(sys.argv[3]), sys.argv[4], sys.argv[5], sys.argv[6])\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryEntryGroup.py", line 687, in main\n    my_entries[entry] = glideFactoryEntry.Entry(entry, startup_dir, glideinDescript, frontendDescript)\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryEntry.py", line 57, in __init__\n    self.jobSubmitAttrs = glideFactoryConfig.JobSubmitAttrs(name)\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryConfig.py", line 300, in __init__\n    lambda s: s,\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryConfig.py", line 102, in __init__\n    ConfigFile.__init__(self, os.path.join("entry_" + entry_name, config_file), convert_function)\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryConfig.py", line 58, in __init__\n    self.load(config_file, convert_function)\n  File "/usr/lib/python3.6/site-packages/glideinwms/factory/glideFactoryConfig.py", line 62, in load\n    with open(fname) as fd:\nFileNotFoundError: [Errno 2] No such file or directory: \'entry_CMSHTPC_T2_PK_NCP_htcondor-ce-2/submit_attrs.cfg\'\n'
```

Notice that compared to the exception in #386 here we have `entry_CMSHTPC_T2_PK_NCP_htcondor-ce-2`, meaning the fix for the wrong location has been applied and it's another type of issue.

**Comment by mambelli** [comment]
The plan is to find an automatic fix for the update.
In the meantime, the workaround in the previous comment will allow successful upgrades

**Comment by mambelli** [comment]
Fixed by #391
